pallets/nfts/src/lib.rs

```diff
@@ -233,25 +262,54 @@ impl<T: Config> NFTs for Pallet<T> {
     type AccountId = T::AccountId;
     type NFTDetails = T::NFTDetails;
     type NFTId = T::NFTId;
+    type NFTSeriesId = T::NFTSeriesId;
 
     fn create(
         owner: &Self::AccountId,
         details: Self::NFTDetails,
+        series_id: Self::NFTSeriesId,
     ) -> result::Result<Self::NFTId, DispatchError> {
-        let id = Total::<T>::get();
-        Total::<T>::put(id.checked_add(&1.into()).ok_or(Error::<T>::NFTIdOverflow)?);
+        let nft_id = Total::<T>::get();
+        Total::<T>::put(
+            nft_id
+                .checked_add(&1.into())
+                .ok_or(Error::<T>::NFTIdOverflow)?,
+        );
+
+        let series_id = if series_id == Default::default() {
+            None
+        } else {
+            Some(series_id)
+        };
```

Comment:
Uhm I think we should either have options in the extrinsic parameters or no options in the struct

Reply:
Removed `Option` altogether.